### PR TITLE
fix: failed to detect now in an error network

### PR DESCRIPTION
**Pull request by HuberTRoy**
[SQN-1297](https://onfinality.atlassian.net/browse/SQN-1297)

### CODE REVIEW

File: public/env.js [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 window.env = {
```
Comment (HZHAIX) [inline public/env.js:4]: We need to ignore this `env.js` file from git.
Btw, your network type is not consistent with the RPC endpoint

File: src/pages/metamask/metamaskView.tsx [hunk 2]
```diff
@@ -26,7 +26,18 @@ const MetaMaskView = () => {
   const [errorMsg, setErrorMsg] = useState('');
 
   useEffect(() => {
-    setNetworkError(error?.name === NetworkError.unSupportedNetworkError);
+    // Web3 may have a bug in production.
+    // a workaround for this.
+    // TODO: further error report for unknow error.
+    if (error) {
+      if (error.name === NetworkError.unSupportedNetworkError) {
```
Comment (HZHAIX) [inline src/pages/metamask/metamaskView.tsx:33]: [Questions] Seems `if (error.name === NetworkError.unSupportedNetworkError)` and `(error.message.includes('Unsupported chain id')` are checking the same error, why need to check separately.

Also both of them are change networkError to `true`, should be abled to combine the condition checking.

Comment (HuberTRoy) [reply src/pages/metamask/metamaskView.tsx:33]: web3-react may have a bug on this. 

If in production env the `error.name` be compressed.

微信截图_20230606132458

Comment (HZHAIX) [reply src/pages/metamask/metamaskView.tsx]: Good catch 👍

**Pull request merged**